src/gui/rss/automatedrssdownloader.cpp

```diff
@@ -597,6 +587,7 @@ void AutomatedRssDownloader::handleFeedCheckStateChange(QListWidgetItem *feed_it
     updateMatchingArticles();
 }
 
+// FIXME: This should return early if any of the filters are invalid
```

Comment:
Ah - see what you mean. Yes - that makes sense, and also in `rssruledownloader.cpp` should check the validity of all fields before attempting to match.